### DISCUSSION TITLE
Can't run VS address sanitizer on GitHub runners

### DISCUSSION THREAD

**Discussion by Reedbeta** [community/community#77047] [Enterprise]
### Select Topic Area

Bug

### Body

I'm having trouble running an executable compiled with Visual Studio with Address Sanitizer enabled, on the Windows runners.

Here's a minimal repro: https://github.com/Reedbeta/gha-asan-test

This is VS's canned hello-world app, which I've modified by [enabling ASan](https://devblogs.microsoft.com/cppblog/addresssanitizer-asan-for-windows-with-msvc/) in the Debug configuration.

Now, running ASan requires `clang_rt.asan_dynamic-x86_64.dll` to be on the PATH. This will be the case when running through a VS developer command prompt, so in my GHA setup I've used the [`setup-msvc-dev`](https://github.com/marketplace/actions/setup-msvc-developer-command-prompt) action to do this.

I've verified that the clang ASan DLL is present in the expected directory on the runner, i.e. `C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\HostX64\x64`, and I've verified that directory is on the PATH, e.g. I can invoke `cl.exe` and it's found.

However, running my hello-world app with ASan on the Windows runner fails immediately with no output and an exit code `-1073741502`, which maps to `0xc0000142` which is `STATUS_DLL_INIT_FAILED`.

I cannot reproduce this on my own machine (Windows 11 10.0.22621.2715, VS Community 17.7.3, same toolchain version 14.37.32822). For me, the app runs without error.

I've spent days trying to debug this and am completely stuck. I suspect something about the configuration of the Windows runner is causing this but I cannot diagnose further on my own.

**Comment by iboB** [comment]
I actually have the same problem.

@Reedbeta did you manage to find a solution?

**Reply by Reedbeta** [reply]
No. I opened [an issue](https://github.com/actions/runner-images/issues/8891#issuecomment-1838995064) with the runner-images repo, but the conclusion was basically ASan's interceptor is barfing on some unexpected content in the Windows DLLs. I don't have time to pursue this further so I've just turned off ASan builds. 😕
